examples/ecies-example.cpp

```diff
@@ -0,0 +1,148 @@
+)";
+
+static const std::string privKeyPem =R"(
+-----BEGIN EC PARAMETERS-----
+BgUrgQQAIg==
+-----END EC PARAMETERS-----
+-----BEGIN EC PRIVATE KEY-----
+MIGkAgEBBDDP/GdwmJa6KFj/R7QJRi7wNNG+viSMos2B+4zIi7fO5BUpIMZObh6/
+ujDnPKFx4SugBwYFK4EEACKhZANiAAQsaP9+pdpv5qbEM5xsNYwCs8GxQw4+iez0
+149XJL0JsiQjx7ec9iLfK0eMmbTczba8zlHhcDEj1wy/luLKUUlbQcGXJ/9xlM7E
+5UJO9ENQcsZN0Zc2XPvhi1YsX4IxfkA=
+-----END EC PRIVATE KEY-----
+)";
+
+static const auto digest = DigestTypes::SHA512;
+static const int digestLength = Hash::getDigestSize(digest);
+static const auto kdfFunc = std::make_shared<X963KDF>(digest);
+/* Dependency injection function, as the key for MAC is generated based on a random ECC key in the ECIES class */
+static const auto macFunc = std::function<std::unique_ptr<MessageAuthenticationCode>(const std::vector<uint8_t>&)>(
+            [](const std::vector<uint8_t> &key) -> std::unique_ptr<MessageAuthenticationCode> {
+    return std::make_unique<mococrw::HMAC>(digest, key);
+});
+
+std::unique_ptr<ECIESEncryptionCtx> buildEncryptionContext()
+{
+    ECIESCtxBuilder encBuilder;
+    auto pubKey = AsymmetricPublicKey::readPublicKeyFromPEM(pubKeyPem);
+
+    return encBuilder.setKDF(kdfFunc)
+            .setMacFactoryFunction(macFunc)
+            .setMacKeySize(digestLength)
+            .buildEncryptionCtx(pubKey);
+}
+
+std::unique_ptr<ECIESDecryptionCtx> buildDecryptionContext(const std::vector<uint8_t> &ephKey)
+{
+    ECIESCtxBuilder decBuilder;
+    auto privKey = AsymmetricPrivateKey::readPrivateKeyFromPEM(privKeyPem, "");
+    std::shared_ptr<AsymmetricKey::Spec> spec = privKey.getKeySpec();
+    auto eccSpec = std::dynamic_pointer_cast<ECCSpec>(spec);
+    if (!eccSpec) {
+        std::cerr << "Given key is no ECC key." << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    auto _ephKey = AsymmetricPublicKey::fromECPoint(eccSpec, ephKey);
+
+    return decBuilder.setKDF(kdfFunc)
+            .setMacFactoryFunction(macFunc)
+            .setMacKeySize(digestLength)
+            .buildDecryptionCtx(privKey, _ephKey);
+}
+
+struct EciesEncryptResult encrypt_ecies()
+{
+    std::unique_ptr<ECIESEncryptionCtx> encCtx = buildEncryptionContext();
+    std::vector<uint8_t> message = utility::fromHex("deadbeef");
+
+    std::vector<uint8_t> encryptedData;
+    try {
+         encCtx->update(message);
+         encryptedData = encCtx->finish();
+    }  catch (MoCOCrWException &e) {
+        std::cerr << "Error encrypting data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    return EciesEncryptResult{
+        encCtx->getEphemeralKey().toECPoint(openssl::EllipticCurvePointConversionForm::uncompressed),
+        encryptedData,
+        encCtx->getMAC(),
+    };
+}
+
+void decrypt_ecies(struct EciesEncryptResult eciesData)
+{
+    /* We need
+     * - mac value
+     * - ephemeral key
+     * - private key
+     */
+
+    /* Use the default values, which should match here. */
```

Comment:
Are you actually using the default values? `buildDecryptionContext` seems very similar to `buildEncryptionContext` to me.